### ISSUE TITLE
Typing inside the text box will thrown an error

### ISSUE THREAD

**Issue by ErkoKnoll** [valor-software/ng2-select#437]
Error is the following:
Error in ./SelectComponent class SelectComponent - inline template:40:19 caused by: value.match is not a function.

**Comment by theunreal** [comment]
Same here..

```
error_handler.js:45 EXCEPTION: value.match is not a functionErrorHandler.handleError @ error_handler.js:45next @ application_ref.js:273schedulerFn @ async.js:82SafeSubscriber.__tryOrUnsub @ Subscriber.js:223SafeSubscriber.next @ Subscriber.js:172Subscriber._next @ Subscriber.js:125Subscriber.next @ Subscriber.js:89Subject.next @ Subject.js:55EventEmitter.emit @ async.js:74onError @ ng_zone.js:120onHandleError @ ng_zone_impl.js:64ZoneDelegate.handleError @ zone.js:196Zone.runGuarded @ zone.js:102NgZoneImpl.runInnerGuarded @ ng_zone_impl.js:72NgZone.runGuarded @ ng_zone.js:236outsideHandler @ dom_events.js:26ZoneDelegate.invokeTask @ zone.js:225Zone.runTask @ zone.js:125ZoneTask.invoke @ zone.js:293
error_handler.js:50 ORIGINAL STACKTRACE:ErrorHandler.handleError @ error_handler.js:50next @ application_ref.js:273schedulerFn @ async.js:82SafeSubscriber.__tryOrUnsub @ Subscriber.js:223SafeSubscriber.next @ Subscriber.js:172Subscriber._next @ Subscriber.js:125Subscriber.next @ Subscriber.js:89Subject.next @ Subject.js:55EventEmitter.emit @ async.js:74onError @ ng_zone.js:120onHandleError @ ng_zone_impl.js:64ZoneDelegate.handleError @ zone.js:196Zone.runGuarded @ zone.js:102NgZoneImpl.runInnerGuarded @ ng_zone_impl.js:72NgZone.runGuarded @ ng_zone.js:236outsideHandler @ dom_events.js:26ZoneDelegate.invokeTask @ zone.js:225Zone.runTask @ zone.js:125ZoneTask.invoke @ zone.js:293
error_handler.js:51 TypeError: value.match is not a function
    at HighlightPipe.transform (select-pipes.js:15)
    at view_utils.js:159
    at AppView._View_SelectComponent8.detectChangesInternal (SelectComponent.ngfactory.js:574)
    at AppView.detectChanges (view.js:272)
    at AppView.detectContentChildrenChanges (view.js:290)
    at AppView._View_SelectComponent7.detectChangesInternal (SelectComponent.ngfactory.js:513)
    at AppView.detectChanges (view.js:272)
    at AppView.detectContentChildrenChanges (view.js:290)
    at AppView._View_SelectComponent1.detectChangesInternal (SelectComponent.ngfactory.js:181)
    at AppView.detectChanges (view.js:272)ErrorHandler.handleError @ error_handler.js:51next @ application_ref.js:273schedulerFn @ async.js:82SafeSubscriber.__tryOrUnsub @ Subscriber.js:223SafeSubscriber.next @ Subscriber.js:172Subscriber._next @ Subscriber.js:125Subscriber.next @ Subscriber.js:89Subject.next @ Subject.js:55EventEmitter.emit @ async.js:74onError @ ng_zone.js:120onHandleError @ ng_zone_impl.js:64ZoneDelegate.handleError @ zone.js:196Zone.runGuarded @ zone.js:102NgZoneImpl.runInnerGuarded @ ng_zone_impl.js:72NgZone.runGuarded @ ng_zone.js:236outsideHandler @ dom_events.js:26ZoneDelegate.invokeTask @ zone.js:225Zone.runTask @ zone.js:125ZoneTask.invoke @ zone.js:293
select-pipes.js:15 Uncaught TypeError: value.match is not a function
```

Seem like the new version is bugged with app error when an user types in the select box @valorkin

**Comment by valorkin** [comment]
@otelnov can you check please?

**Comment by gauzpan** [comment]
@valorkin Facing the same issue. What is the current status of this issue. Is it solved?

**Comment by gauzpan** [comment]
@valorkin Also the problem is there in your demo page also.

**Comment by galvanu** [comment]
Facing the same issue, do you have any idea how to solve this?

**Comment by jppellerin** [comment]
same here.  +1

**Comment by unsafecode** [comment]
@valorkin Has this fix already been published to npm?

**Comment by valorkin** [comment]
Tried to publish today, didn't worked. I will issues and I will publish tomorrow

**Comment by Meligy** [comment]
@valorkin here's a pointer that might help with that problem.  I tried today because I really needed this fix for a demo.

I tried running the following npm script rom the package.json file:

```
    "prepublish": "npm run flow.clean && npm run flow.compile",
```

and it just broke.

Turns out the `flow.clean` breaks it. If you only run the `flow.compile` task, it works just file.

The `flow.clean` task clears the `demo` folder and more, which is required by the `tsconfig.json`.
